### DISCUSSION TITLE
Why doesn't it appear after calling 'image' in my code?

### DISCUSSION THREAD

**Discussion by alfredkamdem** [quarto-dev/quarto-cli#9168] [Q&A]
### Description

Please, what is the problem of this code? Image not appear after compilation.
Thanks for your answers.

 ```{python}
path_image = "C:/Users/Alfred/dashboard/"

if ratio_23_24 > 0:
    image= os.path.abspath(path_image, 'icons_croissance.png')
else:
    image = os.path.join(path_image, 'icons_decroissance.png')
```

::: {.columns}
::: {.valuebox color="white" title="Revenu moyen 2024" label="valuebox3"}
    
::: {style="font-size: large"}
![](image) `{python} result ` 
:::
:::
::::

**Comment by mcanouil** [comment]
> Your question in one sentence (e.g., How do I preview documents with Quarto?)

Could you edit the title of your post for it to be meaningful? (there is an edit button next to the title of your post)

Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

Could you share a small **self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

Note that the main issue for anyone in order to help is that `C:/Users/Alfred/dashboard/` does not exist except on your computer.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Comment by alfredkamdem** [comment]
[fichier_quarto_.md](https://github.com/quarto-dev/quarto-cli/files/14724083/fichier_quarto_.md)

This is small version of my code.

**Reply by mcanouil** [reply]
Please do as requested and share as showed. Thank you for your cooperation.

See https://github.com/quarto-dev/quarto-cli/discussions/9168#discussioncomment-8878743

**Reply by alfredkamdem** [reply]
Why doesn't it appear after calling 'image' in my code?

````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

```{python}
path_image = "C:/Users/Alfred/"
if ratio_23_24 > 0:
    image = os.path.join(path_image, f'icons_decroissance.png')
else:
    image = os.path.join(path_image, 'icons_decroissance.png')
```

### Column
:::: {}
::: {.columns}
::: {.valuebox color="white" title="Revenu moyen 2024" label="valuebox3"} 

::: {style="font-size: large"}
![](image) `{python} result` 
:::
:::

:::
```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```
::::

````

**Reply by mcanouil** [reply]
Are you sure about the format?
Your document seems to be a dashboard.

Why do you have empty divs?

And your document is not reproducible, the first Python code cell will never work for everyone but you.

Note that Quarto main design choice is to have all resources within the project (there is no way around that). Your image path is clearly outside of your project, which might be the reason it is not working.

**Reply by cscheid** [reply]
What do you expect this to produce?

```
![](image) `{python} result` 
```

**Reply by alfredkamdem** [reply]
Excatly. It's  a dashboard. My workdirectory is "C:/Users/Alfred/" and " icons_decroissance.png" is inside this folder.

result = f" + 2.5% par rapport à 2023"
---
title: "Dashboard"
format: 
  dashboard :
    embed-resources : true
theme : custom.scss
scrolling : true
---

**Reply by mcanouil** [reply]
@alfredkamdem take few minutes to properly write/edit your messages or don't reply by email if you don't know how to format your email for it to be legible on GitHub.
Also, your example needs to work at the very minimum for you, if you copy/paste what you shared.

**Comment by alfredkamdem** [comment]
I want to add icon before  + 2.5% par rapport à 2023.
    Le vendredi 22 mars 2024 à 16:22:24 UTC+1, Carlos Scheidegger ***@***.***> a écrit :  
 
 


What do you expect this to produce?
![](image) `{python} result`

**Reply by cscheid** [reply]
Your code creates an image node with URL `image`, and prints the value of `result` after it. Read https://quarto.org/docs/computations/inline-code.html#markdown-output and make sure you understand how to create markdown computationally.
